### ISSUE TITLE
lacking documentation about containers and other concepts

### ISSUE THREAD

**Issue by PerBothner** [swaywm/sway#7323]
I'm playing around with Sway, and am impressed by what it can do. One problem is that the documentation covers the various commands, but it doesn't really explain the concepts. I'm willing to write up something and create a pull request for `sway.5.scd` (or perhaps `sway.1.scd`) but there are big holes in my knowledge. 

I suggest adding a `CONTAINERS` section to  `sway.5.scd`. To start with something like:

Sway organizes its windows as nested containers. A container is a rectangular area of the screen that contains one or more windows or nested sub-containers. Every container has a _layout mode_ which specifies how the sub-containers are displayed in a container: A layout mode is one of: `splith` (the sub-containers are arranged left-to-right); `splitv` (similar but top-to-bottom); `tabbed` (the title bars for all sub-containers are show left-to-right but otherwise only the active sub-container is displayed); `stacked` (similar to `tabbed` but the sub-container titlebars are shown one below each other). You can change the layout mode of a container with the `layout` command.

Each output (screen) has one or more top-level containers: The main tiling area, and zero or more floating (top-level) containers. (A floating container can be split into sub-containers.)

Question: It is unclear if/when a window is also a container. Many places in sway(5) uses the form "[container|window]" which implies a window is not a container. However, other places uses "container" in contexts that suggest a window also a container. Some of these seem to be errors. For example `focus up|right|down|left` is documented to "Moves focus to the next container" but it doesn't - it moves focus to the next _window_.

There seem to be what I call "invisible containers" - containers with a single child (window or sub-containers). These seem to be the reason `$mod+a` (focus parent) will sometimes seem to do nothing - presumably because they selected an invidible container. The `split` command seems to create an invisible container around the current window|container - but only if there isn't already a surrounding invisible container. More use of "indicators" (as in "The color used to indicate where a new view will open") might help.  Perhaps when an invisible container is focused we could change the border color or thickness?

Can an invisible container be nested inside another invisible container? I think the answer should be "no".

Question: sway(5) sometimes talks about "views" - but never defines it. Does it mean window, or container, or either?

**Comment by emersion** [comment]
Yeah, I agree a general overview of Sway's concepts would be quite nice.

> It is unclear if/when a window is also a container

Yeah, a window is a kind of container.

> There seem to be what I call "invisible containers" - containers with a single child

These might be a bug. IIRC we try to eliminate intermediate single-child containers.

> Question: sway(5) sometimes talks about "views" - but never defines it. Does it mean window, or container, or either?

view == window. It's an internal Sway terminology, and TBH "window" is clearer, we should probably try to use the word "window" instead.

**Comment by flber** [comment]
I've been trying to figure out something which is perhaps relevant recently, which is the concept of "window mode"? In the introductory feautures video on the [sway](https://swaywm.org) homepage, the person talking says this:
>  I can also split windows into containers of several windows

whereupon they press the keybinding `super+w`, which puts them into a "window" mode, and then `s`. When they then open a terminal, it opens as a window *in* the newly made container.

This keybinding (and mode), are nowhere to be found in the default config, but would at least to me be super useful! Is this still a feature, and would it be reasonable to reintroduce this to the default config file?

**Comment by mwette** [comment]
This might help: https://i3wm.org/docs/userguide.html

**Comment by WillPower3309** [comment]
I'm hoping to take a stab at this soon, was going to do a high level technical writeup of sway for contributors to swayfx. Where would we like to host such a document? In the repo as `ACHITECTURE.md` or something similar?

**Comment by WhyNotHugo** [comment]
@WillPower3309 The man pages are likely the best place. They themselves refer to 'containers' and 'viewes', so should explain what they are.

**Comment by emersion** [comment]
This issue is about documenting the user-facing concepts of Sway, as opposed to Sway's internals.
